src/DocBlock/TypeExpression.php

```diff
@@ -428,7 +440,50 @@ private function parse(): void
                 \assert(\strlen($this->value) === $index);
 
                 $this->isUnionType = true;
-                $this->typesGlue = $typesGlue;
+
+                if (1 === \count($seenGlues)) {
+                    $this->typesGlue = array_key_first($seenGlues);
+
+                    foreach ($innerValues as $innerValue) {
+                        $this->innerTypeExpressions[] = [
+                            'start_index' => $innerValue['start_index'],
+                            'expression' => $this->inner($innerValue['value']),
+                        ];
+                    }
+                } else {
+                    $glue = null;
+                    foreach (['|', '&'] as $possibleGlue) {
+                        if ($seenGlues[$possibleGlue] ?? false) {
```

Comment:
I would rather like to see:

```suggestion
                        if (in_array($possibleGlue, $seenGlues, true)) {
```

This obviously requires

```diff
-$seenGlues[$matches['glue'][0]] = true;
+$seenGlues[] = $matches['glue'][0];
```

\+ `$seenGlues = array_unique($seenGlues);` somewhere after `while`. I know your code is correct, but IMHO is less readable - by looking at `if ($seenGlues[$possibleGlue] ?? false)` I need to think what's the `$seenGlues`'s content, while with `in_array()` it reads natural, at least for me. But this is only nice to have, I won't force this change.

---

Additionally, if I understand correctly, the order of `['|', '&']` is important because if `&` has been seen, it should be used as a top level glue, while the `|` should be included in the inner expression, right? I am wondering if it's possible to implement it in more explicit way?

Reply:
landed in https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/8210/commits/855a92c71a1fd11bd91e01bfd90f12849f81a6e2